### DISCUSSION TITLE
Only noise when mixing two memory-streams

### DISCUSSION THREAD

**Discussion by veitk** [pschatzmann/arduino-audio-tools#450] [Q&A] [answered]
Hi, 

i now have my header files converted and can play two different sounds. 
now i'm trying to mix those two and looked in your example. 

this is what i did (i leave the config out, it is working):

```
I2SStream i2sStream;
MemoryStream sound1(sound1, sound1_len);
MemoryStream sound2(sound2, sound2_len);

InputMixer<int16_t> mixer;
StreamCopy mixCopier(i2sStream, mixer);

...

mixer.add(sound1);
mixer.add(sound2);

i2sStream.begin(config);

...

void loop() {
  if (!mixCopier.copy2()) {
      Serial.println("finished playing sound");
  }
}

```

But it is mostly noise what i'm hearing. 
Both files are 44.1 16bit and are working perfectly if played alone on a button press. Did i do something wrong?

And one more question: i want to play both sounds on different button-press and mix it on the fly. will this be possible at all? i hope so ;-)

thx!

**Comment by pschatzmann** [comment, marked as answer]
Did you try with the mixCopier.copy().
I am not sure what copy2 does in this case...

**Reply by veitk** [reply]
i didn't even check what the difference is between copy and copy2. since i'm working as an android developer i'm used to classes with the number 2 in it (retrofit2, canvasrenderer2...) which means it is the newer and better version ;-)

it works perfectly now, thanks!

**Reply by pschatzmann** [reply]
Cool: I also added another [example](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/streams-memory_pcm-mixer-audiokit). I also deleted the copy2() method which originally was used to create stereo out of mono data.
